src/checks.rs

```diff
@@ -174,4 +181,13 @@ mod tests {
 
         assert_eq!(expected, run(lines, &skip_checks));
     }
+
+    #[test]
+    fn check_name_list() {
+        let available_check_names = available_check_names();
+        for check in checklist() {
+            let check_name = check.name();
+            assert!(available_check_names.contains(&check_name.to_string()));
```

Comment:
Please, add an integration test for that function as well, because right now it misses the `EndingBlankLine`check.

Reply:
> Please, add an integration test for that function as well, because right now it misses the `EndingBlankLine`check.

Yes indeed!
In this case, we still have to make a strict list of checks (as I wrote [above](https://github.com/dotenv-linter/dotenv-linter/pull/202#issue-422457368)), and when adding new ones, we will need to explicitly update the integration test

But, it seems to me, perhaps, it is conceptually more correct to add the attribute of the last line to the `LineEntry` and check directly inside the check, as in other cases.
Since we have an exception check that takes our logic out.
This, of course, is beyond the scope of this task, but this is an occasion to reflect.

What do you think about it?

Reply:
@mgrachev 
And because it's specific check ("exception"), I doubt between integration test with hardcode strict check list and manually check in module test specific name (`EndingBlankLine`) as in the example below. 
Because it's stands out from the general checks API.

```suggestion
        let available_check_names = available_check_names();
        for check in checklist() {
            let check_name = check.name();
            assert!(available_check_names.contains(&check_name.to_string()));
        }
        let ending_blank_line = ending_blank_line::EndingBlankLineChecker::default();
        assert!(available_check_names.contains(&ending_blank_line.name().to_string()));
    }
```

Reply:
> Yes indeed!
> In this case, we still have to make a strict list of checks (as I wrote [above](https://github.com/dotenv-linter/dotenv-linter/pull/202#issue-422457368)), and when adding new ones, we will need to explicitly update the integration test
> 
> But, it seems to me, perhaps, it is conceptually more correct to add the attribute of the last line to the `LineEntry` and check directly inside the check, as in other cases.
> Since we have an exception check that takes our logic out.
> This, of course, is beyond the scope of this task, but this is an occasion to reflect.
> 
> What do you think about it?

It's an interesting idea. Could you give some example code how you would implement it?

Reply:
> @mgrachev
> And because it's specific check ("exception"), I doubt between integration test with hardcode strict check list and manually check in module test specific name (`EndingBlankLine`) as in the example below.
> Because it's stands out from the general checks API.

Okay, let's leave it like that.

Reply:
> > Yes indeed!
> > In this case, we still have to make a strict list of checks (as I wrote [above](https://github.com/dotenv-linter/dotenv-linter/pull/202#issue-422457368)), and when adding new ones, we will need to explicitly update the integration test
> > But, it seems to me, perhaps, it is conceptually more correct to add the attribute of the last line to the `LineEntry` and check directly inside the check, as in other cases.
> > Since we have an exception check that takes our logic out.
> > This, of course, is beyond the scope of this task, but this is an occasion to reflect.
> > What do you think about it?
> 
> It's an interesting idea. Could you give some example code how would you implement it?

Yes, a little later I will give an example of what I mean

Reply:
> It's an interesting idea. Could you give some example code how would you implement it?

For example, we can add total_number to `LineEntry`
```
pub struct LineEntry {
    pub number: usize,
    pub total_number: usize,
    pub file_path: PathBuf,
    pub raw_string: String,
}
```
And method to `impl LineEntry` :
```
pub fn is_last_line(&self) -> bool {
    self.total_number == self.number
}
```

And in the end we just can add check to `EndingBlankLineChecker`:
`if line.is_last_line() {...}`


It's have one issue: not persistent, if we add line to vector - need update all line entries.

In theory, dynamically adding strings to a vector is not possible, but a large volume of tests needs to be processed, since we do not use the constructor method for `LineEntry`.

Using a boolean flag instead of total_number will simplify the task, you only need to update the last element earlier.
But perhaps total_number may be of more value in terms of check api (can also check first line as example).

We can also calculate the number of lines at the very beginning, which will allow us to allocate a vector with initial capacity, as it:
```
let total_lines = reader.lines().count();
let mut lines: Vec<LineEntry> = Vec::with_capacity(total_lines);
```

Reply:
@mgrachev But, this study is beyond the scope of this task and requires some refactoring.
Therefore, at the moment I am making changes, as talk above, and ask for a re-review of the code.
And this topic can be discussed in a separate issue.

Thanks for good feedback!

Reply:
I like that idea! 👏Thank you! Could you open an issue to implement it?

Reply:
But I would like to move the `total_number` field to the `FileEntry` struct. It will be more correct.

Reply:
Yes of course!  Soon I will create an issue, I also think this option(total_number) is more suitable

Reply:
I have made some refactoring to implement that idea more easily.
https://github.com/dotenv-linter/dotenv-linter/pull/203

Reply:
> I like that idea! :clap: Thank you! Could you open an issue to implement it?

Done #204